entities/Item.entity.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  Entity,
+  Column,
+  PrimaryGeneratedColumn,
+  ManyToOne,
+  JoinColumn,
+} from 'typeorm';
+import {
+  IsDefined,
+  IsUUID,
+  IsOptional,
+  Length,
+  IsDecimal,
+} from 'class-validator';
+import {
+  ApiHideProperty,
+  ApiProperty,
+  ApiPropertyOptional,
+} from '@nestjs/swagger';
+import { Auditable } from './Auditable';
+import { Company } from './Company.entity';
+
+@Entity('item')
+export class Item extends Auditable {
+  @PrimaryGeneratedColumn('uuid')
+  @IsOptional()
+  @IsUUID()
+  @ApiProperty({ description: 'Item Identifier' })
+  id?: string;
+
+  @Column()
+  @IsDefined()
+  @Length(2, 255)
+  @ApiProperty({ description: 'Item Name', minLength: 2, maxLength: 255 })
+  name: string;
+
+  @Column()
+  @IsDefined()
+  @IsDecimal()
+  @ApiPropertyOptional({
+    description: 'Item Cost',
+    type: IsDecimal,
+    nullable: false,
+  })
+  cost?: number;
+
+  @Column()
+  @IsOptional()
+  @Length(2, 255)
+  @ApiPropertyOptional({
+    description: 'Item Description',
+    nullable: true,
+    type: String,
+  })
+  description?: string;
+
+  @Column()
+  @IsOptional()
+  @Length(2, 255)
+  @ApiPropertyOptional({
+    description: 'Item Type',
+    nullable: true,
+    type: String,
+    enum: ['DRINK', 'FOOD'],
+  })
+  type?: string;
```

Comment:
I think this field should be not nullable and have a default 'food'

Reply:
But what if we use this in a future for not only food items @matheo97 , in that case it will be nice to define like a generic type to not force the DB to be related with an specific category of business

Thougts?

Reply:
The other thing is how do I define the default value @matheo97  :trollface: ?